packages/flutter_tools/lib/src/reporting/crash_reporting.dart

```diff
@@ -26,15 +26,61 @@ const String _kStackTraceFileField = 'DartError';
 /// it must be supplied in the request.
 const String _kStackTraceFilename = 'stacktrace_file';
 
+class CrashDetails {
+  CrashDetails({
+    @required this.command,
+    @required this.error,
+    @required this.stackTrace,
+    @required this.doctorText,
+  });
+
+  String command;
+  dynamic error;
+  StackTrace stackTrace;
+  String doctorText;
+}
+
+/// Reports information about the crash to the user.
+class CrashReporter {
+  CrashReporter({
+    @required this.logger,
+    this.gitHubTemplateCreator,
+  });
+
+  final GitHubTemplateCreator gitHubTemplateCreator;
+  final Logger logger;
+
+  /// Prints instructions for filing a bug about the crash.
+  Future<void> informUser(CrashDetails details, File crashFile) async {
+    logger.printError('A crash report has been written to ${crashFile.path}.');
+    logger.printStatus('This crash may already be reported. Check GitHub for similar crashes.', emphasis: true);
+
+    final String similarIssuesURL = GitHubTemplateCreator.toolCrashSimilarIssuesURL('${details.error}');
+    logger.printStatus('$similarIssuesURL\n', wrap: false);
+    logger.printStatus('To report your crash to the Flutter team, first read the guide to filing a bug.', emphasis: true);
+    logger.printStatus('https://flutter.dev/docs/resources/bug-reports\n', wrap: false);
+
+    if (gitHubTemplateCreator != null) {
```

Comment:
Then this null check doesn't need to be here if this class creates the GitHubTemplateCreator.

Reply:
`gitHubTemplateCreator` is supplied as an construction argument, and it could be null. (The google3 subclass will not provide one. Granted, it also will override `informUser`, but in principle overridng `informUser` isn't required.)

Reply:
Oops, I missed the other comment about creating the `GitHubTemplateCreator`.  Okay.